src/lib/services/graphql/graphql-subscriptions-fixture.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-env jest */
+
+import { DocumentNode } from 'graphql';
+import WS from 'jest-websocket-mock';
+import { DeserializedMessage } from 'jest-websocket-mock/lib/websocket';
+import { Observer, Subscription, lastValueFrom, take } from 'rxjs';
+import { ConnectionStatus } from '../../model/connection-status';
+import { GraphqlService } from './graphql.service';
+
+const keyValue = 'temporary_api_key';
+const PORT = 42990;
+const SERVER_URL = `ws://localhost:${PORT}`;
+
+export class GraphQLSubscriptionsFixture {
+  graphqlService = new GraphqlService();
+  server: WS;
+
+  constructor() {
+    this.openServer();
+    jest
+      .spyOn(this.graphqlService as any, 'fetchTemporaryApiKey')
+      .mockResolvedValue(keyValue);
+    jest
+      .spyOn(this.graphqlService as any, 'getServerUrl')
+      .mockReturnValue(SERVER_URL);
+  }
+
+  triggerSubscription(
+    query: DocumentNode | string = 'subscription { baba }',
```

Comment:
I'm not sure how stringent we should be about tests - I think it's probably fine as is.

Though we have to be aware that when writing a new test that calls `triggerSubscription` and passes some new string, then `consumeSubscribeMessage` would not work.

Maybe `consumeSubscribeMessage` could also accept a query param with default value `subscription { baba }`, same as here.